ride_share.rb

```diff
@@ -0,0 +1,272 @@
+        },
+        cost: 25,
+        rider_id: "RD0073",
+        rating: 5
+    },
+    # Ride 3
+    {
+        driver_id: "DR0001",
+        date: {
+            day: "3rd",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 30,
+        rider_id: "RD0015",
+        rating: 4
+    },
+    # Ride 4
+    {
+        driver_id: "DR0003",
+        date: {
+            day: "4th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 5,
+        rider_id: "RD0066",
+        rating: 5
+    },
+    # Ride 5
+    {
+        driver_id: "DR0004",
+        date: {
+            day: "4th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 10,
+        rider_id: "RD0022",
+        rating: 4
+    },
+    # Ride 6
+    {
+        driver_id: "DR0002",
+        date: {
+            day: "4th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 15,
+        rider_id: "RD0013",
+        rating: 1
+    },
+    # Ride 7
+    {
+        driver_id: "DR0003",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 50,
+        rider_id: "RD0003",
+        rating: 2
+    },
+    # Ride 8
+    {
+        driver_id: "DR0002",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 35,
+        rider_id: "RD0066",
+        rating: 3
+    },
+    # Ride 9
+    {
+        driver_id: "DR0004",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 20,
+        rider_id: "RD0073", 
+        rating: 5
+    },
+    # Ride 10
+    {
+        driver_id: "DR0001",
+        date: {
+            day: "5th",
+            month: "Feb",
+            year: 2016
+        },
+        cost: 45,
+        rider_id: "RD0003", 
+        rating: 2
+    }
+]
+
+# Analysis data structure
+drivers = [
+    # Driver 1
+    {
+        driver_id: "DR0001",
+        ride_count: 0,
+        earnings: 0.0,
+        avg_rating: 0.0
+    },
+    # Driver 2
+    {
+        driver_id: "DR0002",
+        ride_count: 0,
+        earnings: 0.0,
+        avg_rating: 0.0
+    },
+    # Driver 3
+    {
+        driver_id: "DR0003",
+        ride_count: 0,
+        earnings: 0.0,
+        avg_rating: 0.0
+    },
+    # Driver 4
+    {
+        driver_id: "DR0004",
+        ride_count: 0,
+        earnings: 0.0,
+        avg_rating: 0.0
+    }
+]
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+# Use an iteration blocks to print the following answers:
+puts "~`~`~`~`~`~`~`~`~ OUTPUT ~`~`~`~`~`~`~`~`~`~"
+# Total number of rides each driver has given
+puts "\nThe number of rides each driver has given."
+for i in (0..drivers.length - 1) do
+    for j in (0..rides.length - 1) do
+        if drivers[i][:driver_id] == rides[j][:driver_id] 
```

Comment:
Instead of using `for` loops here and dealing with indices, it would probably be cleaner to use `each` loops:

```ruby
puts "\nThe number of rides each driver has given."
drivers.each do |driver|
    rides.each do |ride|
        if driver[:driver_id] == ride[:driver_id] 
            driver[:ride_count] += 1
        end
    end
    puts "#{driver[:driver_id]} has given #{driver[:ride_count]} rides."
end
```